### PR TITLE
feat: new react-only step config drawer

### PR DESCRIPTION
**Pull request by AndrasEszes**
https://bitrise.atlassian.net/browse/CI-3376
https://bitrise.atlassian.net/browse/CI-3403

### CODE REVIEW

File: source/javascripts/components/StepConfigDrawer/tabs/PropertiesTab.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  Text,
+  useDisclosure,
+} from '@bitrise/bitkit';
+import { useStepDrawerContext } from '../StepConfigDrawer.context';
+import { isStepLib } from '@/models/Step';
+import { getVersionRemark } from '@/utils/stepVersionUtil';
+
+const PropertiesTab = () => {
+  const { isOpen: showMore, onToggle: toggleShowMore } = useDisclosure();
+  const { cvs, step, availableVersions, selectedVersion } = useStepDrawerContext();
```
Comment (AndrasEszes) [inline source/javascripts/components/StepConfigDrawer/tabs/PropertiesTab.tsx:19]:
```suggestion
  const { cvs, step, title, availableVersions, selectedVersion } = useStepDrawerContext();
```

File: source/javascripts/components/StepConfigDrawer/tabs/PropertiesTab.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+          colorScheme="purple"
+          rel="noreferrer noopener"
+          href={step?.source_code_url}
+          isExternal
+        >
+          <Text>View source code</Text>
+          <Icon name="OpenInBrowser" />
+        </Link>
+      )}
+      <Input defaultValue={step?.title} type="text" label="Name" placeholder="Step name" isRequired />
```
Comment (AndrasEszes) [inline source/javascripts/components/StepConfigDrawer/tabs/PropertiesTab.tsx:53]:
```suggestion
      <Input defaultValue={title} type="text" label="Name" placeholder="Step name" isRequired />
```

File: source/javascripts/components/WorkflowCard/WorkflowCard.tsx [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { Box, Card, CardProps, Collapse, ControlButton, Icon, Text, useDisclosure } from '@bitrise/bitkit';
+import { Fragment } from 'react/jsx-runtime';
```
Comment (AndrasEszes) [inline source/javascripts/components/WorkflowCard/WorkflowCard.tsx:2]:
```suggestion
import { Fragment } from 'react';
```

**Review by zoltanszabo-bitrise — APPROVED** [review]

File: source/javascripts/components/AutoGrowableInput.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    fontFamily,
+    formLabelProps,
+    formControlProps,
+    formHelperTextProps,
+    formErrorMessageProps,
+    onChange,
+    ...rest
+  } = props;
+
+  const setInitialReplicatedValue: LegacyRef<HTMLDivElement> = (e) => {
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/components/AutoGrowableInput.tsx:56]: Why LegacyRef

File: source/javascripts/hooks/useStep.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    },
+  });
+
+  return useMemo(() => {
+    return {
+      cvs,
+      isLoading,
+      step: data?.step,
+      selectedVersion: version,
+      resolvedVersion: version,
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/hooks/useStep.ts:108]: This might be an invalid semver version

File: source/javascripts/hooks/useStep.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+      title: mergedStep.title || cvs,
+      maintainer: stepFromAlgolia?.maintainer,
+      availableVersions: stepFromAlgolia?.availableVersions,
+      isLoading: stepFromAlgolia?.isLoading || stepFromLocalApi?.isLoading,
+      selectedVersion: stepFromAlgolia?.selectedVersion || stepFromLocalApi?.selectedVersion,
+      resolvedVersion: stepFromAlgolia?.resolvedVersion || stepFromLocalApi?.resolvedVersion,
+      icon:
+        mergedStep?.asset_urls?.['icon.svg'] ||
+        mergedStep?.asset_urls?.['icon.png'] ||
+        stepFromAlgolia?.icon ||
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/hooks/useStep.ts:152]: isnt the mergedStep already has everything merged to it, so the icons from the api and local api as well?

**Pull request merged**